README.md

```diff
@@ -57,95 +57,388 @@ Some of the skills/concepts this assignment aims to test (and build upon):
-POST /signup
+`403` for invalid credentials
+```json
 {
-  "displayName": "string",
+  "error": true,
+  "errorMsg": "Invalid Credentials"
+}  
+```
+
+> POST `/signup`
+
+**Request**
+```json
+{
+  "email": "string",
   "username": "string",
   "password": "string",
 }
-returns
+```
+**Response**
+`200` On successful signup, sets a cookie with a session token
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
+```
+
+`403` for existent username
+```json
+{
+  "error": true,
+  "errorMsg": "Username Already In Use"
+}  
+```
+
+> PUT `/post/like?postId=13`
+
+**Request**
+```
+postId : Id of the post the current logged in user wishes to like
+```
+
+**Response**
+`200` On successful like
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
+```
+
+`403` for Non Existent post or user not logged in
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Post Id"
+}  
+```
+
+
+> PUT `/post/comment`
+
+**Request**
+
+```json
+{
+  "postId": "integer",
+  "text": "string"
+}
+```
+
+**Response**
+`200` On successful comment
+```json
+{
+  "error": false,
+  "errorMsg": ""
+}
+```
+
+`403` for user is not logged in or invalid post id
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Post Id"
+}  
+```
+
+> POST `/post/new`
+
+**Request**
+```json
+{
+  "description_text": "string",
+  "src": "string"
+}
+```
+
+_TODO: Decide if we are doing a file upload or just base64 encode the image_
+
+**Response**
+`200` On successful Post
+```json
+{
+  "error": false,
+  "errorMsg": ""
+}
+```
+
+`403` for user is not logged in or image/description invalid
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Description | Invalid Image"
+}  
+```
+
+
+> GET `/user/feed?p=17`
+
+**Request**
+
+```
+p : a number specifying where to begin fetching posts,
+    the server will response with 10 posts from the feed in the range
+    [10*p,10*p+10). If not specified server will respond with ALL posts
+```
+
+**Response**
+`200` On successful fetch
+```json
+[
+  {
+      "id": 1,
+      "meta": {
+          "author": "Nina",
+          "description_text": "My lounge",
+          "published": "Sat Aug 04 2018 20:19:12 GMT+1000 (Australian Eastern Standard Time)",
+          "likes": []
+      },
+      "thumbnail": "man-small.jpg",
+      "src": "man.jpg"
+  },
+  "..."
+]
+```
+
+`403` for invalid position token or non logged in user
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Position Number"
+}  
+```
+
+> PUT `/user/follow?username=barry`
+
+**Request**
+
+```
+username: username of the user the logged in user wishes to follow
+```
+
+**Response**
+`200` On successful Follow
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
+```
 
-200 OK for successful signup.
-403 for user already exists
+`403` for invalid username or user is not logged in
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Username"
+}  
+```
+
+
```

Comment:
Just use same API route for both. Also I reckon again as per above, use standardised id for param.
Let the verbs do the backend work for you.
```
# both here are idempotent. You can't create entry 10 twice, nor can u delete it twice. Same effect.
POST     /user/follow?id=10 # creates a new follow entry
DELETE /user/follow?id=10  # deletes a follow entry
```

Reply:
I honestly thing "deleting" a follow entry is super unintuitive

Reply:
Alright, PUT then